src/System.IO.FileSystem.Watcher.Polling/System/IO/PollingWatcher.cs

```diff
@@ -33,43 +35,47 @@ public class PollingWatcher : IDisposable
         bool _includeSubdirectories;
 
         List<string> _extensionsToWatch;
-        List<string> _directories = new List<string>();
+        string _directory;
         PathToFileStateHashtable _state; // stores state of the directory
         byte _version; // this is used to keep track of removals. // TODO: describe the algorithm
+        static string[] SpecialFiles = new[] { "", ".", ".." };
 
         /// <summary>
         /// Creates an instance of a watcher
         /// </summary>
-        /// <param name="rootDirectory">The directory to watch. It does not support UNC paths (yet).</param>
+        /// <param name="directory">The directory to watch. It does not support UNC paths (yet).</param>
         /// <param name="includeSubdirectories">A bool controlling whether or not subdirectories will be watched too</param>
         /// <param name="pollingIntervalInMilliseconds">Polling interval</param>
-        public PollingWatcher(string rootDirectory, bool includeSubdirectories = false, int pollingIntervalInMilliseconds = 1000)
+        public PollingWatcher(string directory, bool includeSubdirectories = false, int pollingIntervalInMilliseconds = 1000)
         {
+            if (!Directory.Exists(directory))
+                throw new ArgumentException("Directory not found.", nameof(directory));
+
             Tracing = new TraceSource("PollingWatcher");
-            _state = new PathToFileStateHashtable(Tracing); 
+            _state = new PathToFileStateHashtable(Tracing);
             _pollingIntervalInMilliseconds = pollingIntervalInMilliseconds;
             _includeSubdirectories = includeSubdirectories;
-            _directories.Add(ToDirectoryFormat(rootDirectory));
+            _directory = directory;
         }
 
         /// <summary>
         /// Extensions to watch
         /// </summary>
-        /// <param name="extension">for examople "txt", "doc", etc., i.e don't use wildcards</param>
+        /// <param name="extension">for example "txt", "doc", etc., i.e don't use wildcards</param>
         /// <remarks>
         /// By default all extensions are watched. Once this method is called, only extensions in added are watched.
         /// </remarks>
         public void AddExtension(string extension)
```

Comment:
We should eventually remove this and make it `Filters`. We didn't have a non-allocating matcher when this was first written.

Reply:
This is coming in my next PR which aims to bring the `PollingFileSystemWatcher` API in closer alignment with `FileSystemWatcher`.